lib/datasource/maven/index.js

```diff
@@ -11,8 +11,54 @@ module.exports = {
   getPkgReleases,
 };
 
+const cacheNamespace = 'datasource-maven';
+
+async function getPkgReleases({ lookupName, registryUrls, groupName, deps }) {
+  let maxGroupVersion = null;
+  if (groupName) {
+    const groupDeps = deps
+      .filter(dep => dep.groupName === groupName)
+      .map(dep => ({
+        lookupName: dep.depName,
+        registryUrls: dep.registryUrls,
+      }));
+    for (let i = 0; i < groupDeps.length; i += 1) {
+      const dep = groupDeps[i];
+      const depReleases = await getCachedPkgReleases(dep);
+      const depVersions = depReleases.releases.map(({ version }) => version);
+      const depMaxVersion = depVersions.reduce((x, y) =>
+        compare(x, y) === -1 ? y : x
+      );
+      if (!maxGroupVersion || compare(maxGroupVersion, depMaxVersion) === 1) {
+        maxGroupVersion = depMaxVersion;
+      }
+    }
+  }
+
+  const result = await getCachedPkgReleases({ lookupName, registryUrls });
+
+  if (maxGroupVersion) {
+    const { releases } = result;
+    return {
+      ...result,
+      releases: releases.filter(
+        ({ version }) => compare(version, maxGroupVersion) !== 1
+      ),
+    };
+  }
+
+  return result;
+}
+
 // eslint-disable-next-line no-unused-vars
-async function getPkgReleases({ lookupName, registryUrls }) {
+async function getCachedPkgReleases({ lookupName, registryUrls }) {
```

Comment:
Is the refactoring of this mostly about efficiency and not looking up a dep twice?

Reply:
Yes, and also about decoupling cache and grouping